python/ray/_private/services.py

```diff
@@ -186,7 +186,8 @@ def get_address_info_from_redis_helper(redis_address,
         client_node_ip_address = client_info["NodeManagerAddress"]
         if (client_node_ip_address == node_ip_address
                 or (client_node_ip_address == "127.0.0.1"
-                    and redis_ip_address == get_node_ip_address())):
+                    and redis_ip_address == get_node_ip_address())
+                or client_node_ip_address == redis_ip_address):
```

Comment:
If we call `ray.init` from worker node (not head node), this will return the wrong `address info`. We should return the given worker node address info not the head node address info. @roireshef @ericl